### DISCUSSION TITLE
Understanding Weblate's Git Management Mechanism for String States

### DISCUSSION THREAD

**Discussion by vanlinh2204** [WeblateOrg/weblate#10663] [Q&A]
Hello Weblate team,

I'm seeking clarity on the Git management mechanism within Weblate, specifically related to the states assigned to string messages in a bundle (e.g., fuzzy, translated, approved).

My question revolves around whether the state of a string in Weblate is directly tied to the process of updating the string on the Git repository. Here's a scenario I encountered:

I edit a string with the state "needs edit" in Weblate.
Even after the edit, the string retains the "needs edit" state.
However, in the repository management interface, I can observe a single change, and I can indeed commit and push the modified string to the remote repository.
The core of my question is: Does Weblate have an additional mechanism that influences when a Git change is made? For instance, is there a way to ensure that a Git change occurs only when the state of a string is "approved" or another specific state?

Any insights into Weblate's Git integration and its handling of string states would be highly appreciated. Thank you!

**Comment by nijel** [comment]
Formats which store state (for example xliff) the state is updated and parsed from git.

For others, the extended state is preserved in Weblate as long as string doesn't change.
